src/vm/eventpipefile.cpp

```diff
@@ -138,8 +125,36 @@ EventPipeFile::EventPipeFile(StreamWriter *pStreamWriter, EventPipeSerialization
     QueryPerformanceCounter(&m_lastSortedTimestamp);
 #endif
 
-    // Write the first object to the file.
-    m_pSerializer->WriteObject(this);
+
+}
+
+void EventPipeFile::InitializeFile()
+{
+    CONTRACTL
+    {
+        THROWS;
+        GC_TRIGGERS;
+        MODE_PREEMPTIVE;
+        PRECONDITION(m_pStreamWriter != nullptr);
+        PRECONDITION(m_pSerializer == nullptr);
+    }
+    CONTRACTL_END;
+
+    bool fSuccess = true;
+    if (m_format >= EventPipeSerializationFormat::NetTraceV4)
+    {
+        const char* pHeader = "Nettrace";
+        uint32_t bytesWritten = 0;
+        fSuccess = m_pStreamWriter->Write(pHeader, 8, bytesWritten) && bytesWritten == 8;
+    }
+    if (fSuccess)
+    {
+        // Create the file stream and write the FastSerialization header.
+        m_pSerializer = new (nothrow) FastSerializer(m_pStreamWriter);
```

Comment:
Why do you want to use `nothrow` here?

Reply:
We don't!  For some reason I was thinking the `fSuccess` was a null check 😫